pkg/api/types.go

```diff
@@ -466,6 +470,30 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
+type RBDVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
+	EndpointsName string `json:"endpoints"`
+	// Required: RBDImage is the rados image name
+	RBDImage string `json:"image"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	// TODO: how do we prevent errors in the filesystem from compromising the machine
+	FSType string `json:"fsType,omitempty"`
+	// Optional: RBDPool is the rados pool name,default is rbd
+	RBDPool string `json:"pool"`
+	// Optional: RadosUser is the rados user name, default is admin
+	RadosUser string `json:"user"`
+	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
+	Keyring string `json:"keyring"`
+	// Optional: Secret is the authentication secret for RBDUser, default is empty.
+	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
```

Comment:
> Thanks @liewegas.
> 
> @rootfs is telling me that an admin can provision a Ceph volume of, say, 5gb size, get an ID for it, and mount it via that ID in the Kubelet. So
> long as pod authors cannot traverse outside of that volume and read other people's data, then I think we're covered for a PV. This requires no
> Secret data. Just an ID for the Ceph volume that we can post as part of the PV.

Right.  If the kubelet only "sees" the bind-mounted fs sitting on the 
block device itself (or even the block device) they can't see anything 
else.

Whoever sees the ceph secret, of course, can access all sorts of other 
stuff, but I'm assuming that is not exposed to the user?  I haven't 
followed the namespacing discussion but I'm hoping it doesn't mean that 
any user can access a ceph secret in the global namespace...

> When the user gets a volume via their PVClaim, they won't know or care that it's Ceph (or NFS or GCE or whatever).
> 
> Is this correct?

That sounds right to me!